src/libraries/System.Text.Json/src/System.Text.Json.csproj

```diff
@@ -18,9 +18,6 @@ The System.Text.Json library is built-in as part of the shared framework in .NET
 
   <!-- DesignTimeBuild requires all the TargetFramework Derived Properties to not be present in the first property group. -->
   <PropertyGroup>
-    <!-- For the inbox library (that is shipping with the product), this should always be true. -->
-    <!-- BUILDING_INBOX_LIBRARY is only false when building the netstandard compatible NuGet package. -->
-    <DefineConstants Condition="$([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) == '.NETCoreApp'">$(DefineConstants);BUILDING_INBOX_LIBRARY</DefineConstants>
```

Comment:
FWIW this wasn't true anyway anymore as the net6.0 tfm exclusively ships out-of-box and the net7.0 tfm also ships out-of-box. So agreed that this should be cleaned-up.